### PR TITLE
zacrichmond/support staging deployments

### PR DESCRIPTION
**Pull request by zacrichmond**
Added support for ingesting staging deployment events

### CODE REVIEW

File: src/entry/webtriggers/gitlab-event-handlers/handle-deployment-event.test.ts [hunk 1]
```diff
@@ -37,25 +35,24 @@ jest.mock('../../../utils/has-deployment-after-28days');
 const mockedGetEnvironments = mocked(getEnvironments);
 const mockedGetDeployment = mocked(getDeployment);
 const mockedSendEventsToCompass = mocked(sendEventToCompass);
-const mockedGetRecentDeployments = mocked(getRecentDeployments);
-const mockedInsertMetricValues = mocked(insertMetricValues);
-const mockedHasDeploymentAfter28Days = mocked(hasDeploymentAfter28Days);
```
Comment (zacrichmond) [inline src/entry/webtriggers/gitlab-event-handlers/handle-deployment-event.test.ts:42]: None of these calls are in the current `handle-deployment-event` path. Removed them from the tests

**Review by gtchuinkam — APPROVED** [review]
Thanks for the refactors and updating tests!

**Comment by gtchuinkam** [comment]
Looks like the linter found a type error

**Comment by zacrichmond** [comment]
I've fixed the issue, this should be ready review again :D

File: src/services/compute-event-and-metrics/get-recent-deployments.ts [hunk 2]
```diff
@@ -1,17 +1,58 @@
...
   projectId: number,
   projectName: string,
-  projectEnvironments?: Environment[],
+  environmentTiers: EnvironmentTier[],
 ): Promise<DataProviderDeploymentEvent[]> => {
-  const environments = projectEnvironments || (await getProjectEnvironments(projectId, groupToken));
+  // get all project environments
+  const projectEnvironments = await getProjectEnvironments(projectId, groupToken);
+
+  const deploymentEventsByEnvironmentTier = environmentTiers.map(async (environmentTier) => {
```
Comment (bielaski) [inline src/services/compute-event-and-metrics/get-recent-deployments.ts:18]: We could probably use flatMap here to remove the need for the flat on line 34

File: src/services/compute-event-and-metrics/get-recent-deployments.ts [hunk 3]
```diff
@@ -1,17 +1,58 @@
...
+    // get a list of deployments for each project environment belonging to the environmentTier
+    const getDeploymentsPromises = projectEnvironments.reduce<Promise<Deployment[]>[]>(
```
Comment (bielaski) [inline src/services/compute-event-and-metrics/get-recent-deployments.ts:20]: My slight preference here is to do something like 
`const getDeploymentsPromises = await Promise.all(projectEnvironments.filter((env) => env.tier in environmentTiers).flatMap((env) => getRecentDeployments(groupToken, projectId, getDateInThePast(), env.name)))`

Comment (zacrichmond) [reply src/services/compute-event-and-metrics/get-recent-deployments.ts:20]: I've adjusted it to use `.filter(...).map(...)` instead

**Comment by zacrichmond** [comment]
I've made some changes which should hopefully simplify the code.

**Review by bielaski — APPROVED** [review]

**Pull request merged**